Route::view now accepts blank $view parameter

faabiopontes opened the pull request:
- The second parameter $view now can be null
- When the parameter $view is null te $uri will be searched as view
- No tests were found for Route::view, don't know what to do next

Test Scenario: I need to access my view `test.blade.php` with no data.
For access it, I have to call `Route::view('test','test');`
With this pull-request we can just `Route::view('test');`

faabiopontes committed:
Route::view now accepts blank $view parameter
- The second parameter $view now can be null
- When the parameter $view is null te $uri will be searched as view
- No tests were found for Route::view, don't know what to do next

GeorgeHanson commented:
Might be worth adding tests to this.

How does it handle slashes? You usually denote a view using a "." for directories. E.g: templates.dashboard. If you are defaulting to the URI, it would be searching for "templates/dashboard"? I'm not sure how this would work, I haven't delved into it too much, but it might be something worth considering

GeorgeHanson commented on src/Illuminate/Routing/Router.php:
```diff
@@ -251,10 +251,10 @@ public function redirect($uri, $destination, $status = 301)
     {
         return $this->match(['GET', 'HEAD'], $uri, '\Illuminate\Routing\ViewController')
-                ->defaults('view', $view)
+                ->defaults('view', $view ? $view : $uri)
```
You could probably simplify this. `$view ?: $uri`

faabiopontes reviewed (commented):
You're right! I learned this after the PR.. Didn't know how to change it without doing another PR.

bencomeau commented:
Just FYI: there are view route tests located at `tests\Integration\Routing\RouteViewTest.php`!

faabiopontes committed:
REF Ternary Simplify

GeorgeHanson commented:
On another note from a developers point of view, I prefer not to have this change. Reason being is that the shortcut can only be used if you don't want to pass data into the view as there is no way of passing data to the view while using this shortcut

robclancy commented:
It's not uncommon for Laravel to do something like `is_array` for shortcuts like this. So...

```php
if (is_array($view)) {
    $data = $view;
    $view = null;
}
```

Then you can do the shortcut with data as the second argument.

faabiopontes committed:
REF Function toViewFormat created
- The function receives a $uri and returns the corresponding view string format
- We now check if the number of parameters is 2 and the $view parameter is  a  array, so we use it as the $data variable

faabiopontes commented:
Added condition for number of parameters and atribution to $data variable thanks to @robclancy

GeorgeHanson commented on src/Illuminate/Routing/Router.php:
```diff
@@ -251,13 +251,29 @@ public function redirect($uri, $destination, $status = 301)
+
         return $this->match(['GET', 'HEAD'], $uri, '\Illuminate\Routing\ViewController')
-                ->defaults('view', $view)
+                ->defaults('view', $view ?: $this->toViewFormat($uri))
```
I'm not sure if this needs to be separated out into its own method, seems like something un-necessary for something simple?

faabiopontes replied:
I added as a own method because I could have forgot some rule on the conversion of URI to view string

Dylan-DPC commented on src/Illuminate/Routing/Router.php:
```diff
@@ -251,13 +251,29 @@ public function redirect($uri, $destination, $status = 301)
      * @return \Illuminate\Routing\Route
      */
-    public function view($uri, $view, $data = [])
+    public function view($uri, $view = null, $data = [])
```
default parameters should always be last in order, how do you expect someone to call this if he wants the default value? 
Also this can be considered a breaking change

robclancy replied:
... they are last?

Dylan-DPC replied:
My bad 😁

taylorotwell commented:
Would rather just let the view be specified explicitly.
